### DISCUSSION TITLE
communicating with hub (brick?)

### DISCUSSION THREAD

**Discussion by brhodewalt** [pybricks/support#393] [Q&A]
First, "hub" or "brick"?

Next, how can I communicate in real time between a PC or Mac (via USB)?

Related question: Can I upload custom media files to the hub/brick?

Thanks!

**Comment by dlech** [comment]
For Powered Up, we usually say "hub" for EV3/NXT, we say "brick". But they are pretty much interchangeable. Which hub/brick are you using?

Pybricks currently only supports communication via Bluetooth Low Energy on the newer Powered Up devices. EV3 is a bit different.

**Comment by brhodewalt** [comment]
I’m using EV3. Sorry that wasn’t clear.

Sent from my iPhone

On Jul 16, 2021, at 11:34 AM, David Lechner ***@***.***> wrote:

﻿ CAUTION: This email originated from outside of the organization. Do not click links or open attachments unless you recognize the sender and know the content is safe.
________________________________

For Powered Up, we usually say "hub" for EV3/NXT, we say "brick". But they are pretty much interchangeable. Which hub/brick are you using?

Pybricks currently only supports communication via Bluetooth Low Energy on the newer Powered Up devices. EV3 is a bit different.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/pybricks/support/discussions/393#discussioncomment-1014459>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALLPK7XS3CTBCOGNN6Q3D33TYB3R3ANCNFSM5AP6YGKA>.






CONFIDENTIALITY STATEMENT: The information in this e-mail inclusive of any attachment(s) is covered by the Electronic Communications Privacy Act, 18 USC SS 2510-2521 and is legally privileged. It is intended only for the attention and use of the named recipient. If you are not the intended recipient, you are not authorized to retain, disclose, copy or distribute the message and/or any of its attachments. If you received this e-mail in error, please notify me by return email and delete this message.

**Reply by dlech** [reply]
In that case, you can use the USB as a network connection and use basic network sockets or anything that builds on top of that (HTTP, MQTT, etc.).

As for media files, the easiest thing to do is just include them in you project folder and it will be included when you download and run (assuming you are using the ev3dev VS Code extension). If you have files that you want to share with may projects, you could create a new folder like `/home/robot/media/` and copy all of your files there.

**Comment by brhodewalt** [comment]
Thank you. I see your advice about including media files in the folder is included in the documentation itself. I'm having a bit of trouble keeping straight which Python/EV version/OS I'm reading about at any one moment.

Bruce Rhodewalt (he/him/his)
RMHS Math Teacher
Google Educator
***@***.***
http://rhodewalt.com/room608.html


________________________________________
